### ISSUE TITLE
chalice deploy  returned non-zero exit status 1

### ISSUE THREAD

**Issue by whatsdis** [awslabs/chalice#69]
so this time I used the virtualenv, and followed the instructions exactly. Ubuntu 16.04, Python 2.7.6, pip 8.1.2

```
(chalice) vagrant@vagrant-ubuntu-trusty-64:/vagrant/dev/helloguys$ chalice deploy
Initial creation of lambda function.
Creating role
Creating deployment package.
Traceback (most recent call last):
  File "/usr/local/bin/virtualenv", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/virtualenv.py", line 711, in main
    symlink=options.symlink)
  File "/usr/local/lib/python2.7/dist-packages/virtualenv.py", line 924, in create_environment
    site_packages=site_packages, clear=clear, symlink=symlink))
  File "/usr/local/lib/python2.7/dist-packages/virtualenv.py", line 1369, in install_python
    os.symlink(py_executable_base, full_pth)
OSError: [Errno 71] Protocol error
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/chalice/bin/chalice", line 11, in <module>
    sys.exit(main())
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/chalice/cli/__init__.py", line 203, in main
    return cli(obj={})
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/chalice/cli/__init__.py", line 123, in deploy
    d.deploy(ctx.obj)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/chalice/deployer.py", line 203, in deploy
    self._deploy_lambda(config)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/chalice/deployer.py", line 218, in _deploy_lambda
    function_arn = self._first_time_lambda_create(config)
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/chalice/deployer.py", line 261, in _first_time_lambda_create
    config['project_dir'])
  File "/home/vagrant/.virtualenvs/chalice/local/lib/python2.7/site-packages/chalice/deployer.py", line 631, in create_deployment_package
    subprocess.check_output(['virtualenv', venv_dir])
  File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['virtualenv', '/vagrant/dev/helloguys/.chalice/venv']' returned non-zero exit status 1
```

**Comment by jamesls** [comment]
The `Protocol error` message is interesting.  Do you happen to be on windows for the host OS?  If so, is seems similar to this issue: https://github.com/gratipay/gratipay.com/issues/2327, which has some possible suggestions for how to fix this issue that I could try.

Could you confirm though that you're using windows as your host OS?

**Comment by whatsdis** [comment]
yes, but getting the same error on ubuntu 16.04 OS (dual boot)

**Comment by jamesls** [comment]
I believe this is fixed in #388.  We no longer use virtual environments when creating deployment package so we shouldn't run into this issue any more.  If anyone's still seeing this issue let me know.
